### PR TITLE
Filter by category (milestone 4)

### PR DESCRIPTION
**Pull request by AlfredoElizarraras**
### This pull request includes the following:
- [x] Enhance the Redux store by adding a filter object.
- [x] Add CategoryFilter component
- [x] Add it to BookList component and Connect it to the Redux store

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @AlfredoElizarraras 
### You have done well so far :clap: However, you have a few issues to fix:-
- [ ] Can you improve the design of the app, add some spacing make the components be easily visible.
- [ ] When a user saves the book, the title somewhat saves with some funny characters, different from what the user did input. Refer to the image below.
design
### Make the changes and submit for another code review. Good luck.
- Happy coding!

**Comment by AlfredoElizarraras** [comment]
### Hello @rOluochKe :wave: 

#### Thanks for your review :pray: 
#### I will implement the suggested changes and submit them for another review. :raised_hands: 

However, I thought I should work on the app design in the next milestone
and well about the characters that are added to the title are part of the id column
I'm using the UUID package to generate the book's id.

File: src/components/BookForm/BookForm.js [hunk 1]
```diff
@@ -49,8 +50,10 @@ class BookForm extends React.Component {
             value={title}
           />
         </label>
-        <label htmlFor="categories">
-          Choose a car:
+        <label htmlFor="categories" className="flex flex-column">
+          <span>
+            Choose a car:
```
Comment (Torres-ssf) [inline src/components/BookForm/BookForm.js]: - Why choose a car :thinking:?  Shouldn't it be cat or category?

**Review by Torres-ssf — APPROVED** [review]
### Milestone approved :+1: 
Hi @AlfredoElizarraras 
Great job on this milestone
Good luck with the next one :smile: 

Happy coding :computer: 
**Sergio Torres**

**Comment by AlfredoElizarraras** [comment]
### Hello @Torres-ssf  :wave: 

#### Thanks for your review and approval :pray: 
#### You are alright, I fix now the car typo :laughing: :raised_hands:

**Pull request merged**